.github/workflows/wheels.yml

```diff
@@ -37,170 +32,144 @@ permissions:
         with:
-          auto-update-conda: true
-          # Really doesn't matter what version we upload with
-          # just the version we test with
-          python-version: '3.10'
-          channels: conda-forge
-          channel-priority: true
-          # mamba fails to solve, also we really don't need this since we're just installing python
-          # mamba-version: "*"
+          name: sdist
+          path: ./dist
 
-      - name: Build sdist
-        run: |
-          pip install build
-          python -m build --sdist
-      - name: Test the sdist
-        shell: bash -el {0}
-        run: |
-          # TODO: Don't run test suite, and instead build wheels from sdist
-          # by splitting the wheel builders into a two stage job
-          # (1. Generate sdist 2. Build wheels from sdist)
-          # This tests the sdists, and saves some build time
-          python -m pip install dist/*.gz
-          pip install hypothesis>=6.46.1 pytest>=7.0.0 pytest-xdist>=2.2.0 pytest-asyncio>=0.17
-          cd .. # Not a good idea to test within the src tree
-          python -c "import pandas; print(pandas.__version__);
-          pandas.test(extra_args=['-m not clipboard and not single_cpu and not slow and not network and not db', '-n 2']);
-          pandas.test(extra_args=['-m not clipboard and single_cpu and not slow and not network and not db'])"
-      - uses: actions/upload-artifact@v3
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.12.3
         with:
-          name: sdist
-          path: ./dist/*
+          package-dir: ./dist/${{ needs.build_sdist.outputs.sdist_file }}
+        env:
+          CIBW_BUILD: ${{ matrix.python[0] }}-${{ matrix.buildplat[1] }}
 
-      - name: Install anaconda client
-        if: ${{ success() && (env.IS_SCHEDULE_DISPATCH == 'true' || env.IS_PUSH == 'true') }}
+      - name: Set up Python
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-name: wheel-env
+          create-args: >-
+            python=${{ matrix.python[1] }}
+            anaconda-client
+            wheel
+          cache-downloads: true
+          cache-environment: true
+
+      - name: Validate wheel RECORD
         shell: bash -el {0}
+        run: for whl in $(ls wheelhouse); do wheel unpack wheelhouse/$whl -d /tmp; done
+
+      # Testing on windowsservercore instead of GHA runner to fail on missing DLLs
+      - name: Test Windows Wheels
+        if: ${{ matrix.buildplat[1] == 'win_amd64' }}
+        shell: pwsh
         run: |
-          conda install -q -y anaconda-client
+          docker pull python:${{ matrix.python[1] }}-windowsservercore
+          docker run python:${{ matrix.python[1] }}-windowsservercore python -m `
+            pip install pytz six numpy python-dateutil tzdata>=2022.1 `
+            hypothesis>=6.46.1 pytest>=7.0.0 pytest-xdist>=2.2.0 pytest-asyncio>=0.17
+          $LAST_DEPS_ID = docker ps -lq
+          docker commit $LAST_DEPS_ID test_deps

```

Comment:
Is there any way to do this all in one command (like the old 32-bit Linux workflow https://github.com/pandas-dev/pandas/blob/778ab8234059059d77eb5d71442c959f8d15c1e0/.github/workflows/32-bit-linux.yml#L35-L46)?

Otherwise I would prefer everything stay in the ``test_wheels_windows.bat`` file.

Reply:
I was close to getting this to work in one command but I was having trouble getting the Windows Runner's `shell: bash` (git bash) to mount a volume to the container. I tried these various syntaxes

https://github.com/pandas-dev/pandas/pull/53087/commits/9fe2bfa5882598793a914b8f2c401a7b61918de3
https://github.com/pandas-dev/pandas/pull/53087/commits/42a90abb2686231ba1958281599e894169904296
https://github.com/pandas-dev/pandas/pull/53087/commits/3ae62a2c8762f100dd79ac3642394e9dc82a6b1e
https://github.com/pandas-dev/pandas/pull/53087/commits/0403f691569dc7287b6ad0ff8aa8fcb404b394e2

Do you happen to know the correct syntax?

Reply:
Yeah, I dunno how to do this either, which is why I decided to keep everything in the separate bat file (after messing around with it a bunch).

Reply:
I finally was able to get this in one command so this should be ready